eclair-rpc/src/test/scala/org/bitcoins/eclair/rpc/EclairRpcClientTest.scala

```diff
@@ -32,89 +32,150 @@ class EclairRpcClientTest extends AsyncFlatSpec with BeforeAndAfterAll {
 
   val logger: Logger = BitcoinSLogger.logger
 
-  val bitcoindRpcClient: BitcoindRpcClient =
+  val bitcoindRpcClientF: Future[BitcoindRpcClient] =
     BitcoindRpcTestUtil.startedBitcoindRpcClient()
 
-  lazy val EclairNodes4(firstClient, secondClient, thirdClient, fourthClient) = {
-    val EclairNodes4(first, second, third, fourth) =
-      EclairRpcTestUtil.createNodeLink(bitcoindRpcClient)
-    clients ++= List(first, second, third, fourth)
-    EclairNodes4(first, second, third, fourth)
+  lazy val eclairNodesF: Future[EclairNodes4] = {
+    bitcoindRpcClientF.flatMap { bitcoindRpcClient =>
+      val nodesF = EclairRpcTestUtil.createNodeLink(bitcoindRpcClient)
 
+      val addedF = nodesF.map { nodes =>
+        clients ++= List(nodes.c1, nodes.c2, nodes.c3, nodes.c4)
+      }
+
+      addedF.flatMap(_ => nodesF)
+    }
   }
 
-  lazy val (client, otherClient) = {
-    val (c1, c2) = EclairRpcTestUtil.createNodePair(Some(bitcoindRpcClient))
-    clients += c1
-    clients += c2
-    (c1, c2)
+  lazy val firstClientF = eclairNodesF.map(_.c1)
+
+  lazy val secondClientF = eclairNodesF.map(_.c2)
+
+  lazy val thirdClientF = eclairNodesF.map(_.c3)
+
+  lazy val fourthClientF = eclairNodesF.map(_.c4)
+
+
+
+  lazy val clientOtherClientF = {
+
+    //use second and third client above since they
+    //aren't really being used in the tests that use eclairNodesF
+    secondClientF.flatMap(s => thirdClientF.map(t => (s,t)))
   }
 
+  lazy val clientF = clientOtherClientF.map(_._1)
+  lazy val otherClientF = clientOtherClientF.map(_._2)
+
   private val clients =
     Vector.newBuilder[EclairRpcClient]
 
   override def beforeAll(): Unit = {
     // make sure we have enough money open channels
-    bitcoindRpcClient.generate(200)
+    //not async safe
+    bitcoindRpcClientF.flatMap(_.generate(200))
```

Comment:
perhaps we want do define the `bitcoindRpcClientF` in such a way that the future only completes when the `generate` call is done? I.e. move the `flatMap` to `generate` up to the definition of this val?